src/Updatedge.net/Services/V1/AvailabilityService.cs

```diff
@@ -18,8 +19,15 @@ public AvailabilityService(string baseUrl, string apiKey) : base(baseUrl, apiKey
         public async virtual Task<OkApiResult<List<WorkerAvailabilityIntervals>>> GetAvailabilityDailyAsync
             (DateTimeOffset start, DateTimeOffset end, int daysToRepeat, IEnumerable<string> workerIds)
         {
-            try
+            try 
             {
+                // Validate start and end dates
+                if (start > end) throw new ApiWrapperException("Start date cannot be after end date");
+                if (end > start.AddHours(24)) throw new ApiWrapperException("End date must be within 24 hours of Start date (inclusive).");
+
+                // Validate worker Ids
+                if (workerIds.Count() == 0) throw new ApiWrapperException("You must supply at least one worker id.");
+

```

Comment:
Just a thought, can we bung these strings in a resource file?  If it's shared across projects that would also be useful future-wise.

Reply:
ok no probs. I think be changing this quite a bit once I've got the validation class in place, so I can add a resource file then.